closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -2306,6 +2344,145 @@ Java_jdk_crypto_jniprovider_NativeCrypto_ECNativeGF2m
     return OSSL_ECGF2M;
 }
 
+static int
+getArrayFromBN(const BIGNUM *bn, unsigned char *out, int len)
+{
+    int ret = -1;
+    int retLen = 0;
+    int size_diff = 0;
+    unsigned char *intermediate;
+
+    int bn_len_bits = (*OSSL_BN_num_bits)(bn);
+    int bn_len = (bn_len_bits + 7) / 8;
+
+    if (bn_len < len) {
+        intermediate = malloc(bn_len);
+        if (NULL == intermediate) {
+            return 0;
+        }
+        retLen = (*OSSL_BN_bn2bin)(bn, intermediate);
+        size_diff = len - bn_len;
+        memset(out, 0x00, size_diff);
+        memcpy(out + size_diff, intermediate, (size_t) bn_len);
+        free(intermediate);
+    } else {
+        retLen = (*OSSL_BN_bn2bin)(bn, out);
+    }
```

Comment:
This should simply fail if `bn_len > len` (rather than write beyond the end of the array.
You should perhaps consider using `BN_bn2binpad()` which applies more directly here.

Reply:
`bn_len` is never supposed to be greater than `len`. The only reason it might be smaller is because the struct will strip any zeroes at the beginning of the number. So the else statement is for `bn_len == len`. I could add another segment in case something goes horribly wrong, even though it's not likely since the length parameters are coming from the curve definition.

As far as the `BN_bn2binpad()` method, it was the one I initially used, but it is not available in earlier versions, so I had to write my own padding implementation.